test/e2e_node/topology_manager_test.go

```diff
@@ -99,6 +101,46 @@ func detectThreadPerCore() int {
 	return threadCount
 }
 
+// for now we only look for pairs of nodes with minimum distance, we also assume that distance table is symmetric.
+func getMinRemoteDistanceForNode(nodeToDistances map[int][]int) int {
+	var minDistance int = 99
+	for myID, distances := range nodeToDistances {
+		if len(distances) == 0 {
+			continue
+		}
+		for nodeID, distance := range distances {
+			// it'll always equal 10, which means local access.
+			if myID == nodeID {
+				continue
+			}
+			if distance < minDistance {
+				minDistance = distance
+			}
+		}
+	}
+
+	return minDistance
+}
+
+func detectNUMADistances(numaNodes int) map[int][]int {
+	nodeToDistances := make(map[int][]int)
+	for i := 0; i < numaNodes; i++ {
+		outData, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("cat /sys/devices/system/node/node%d/distance", i)).Output()
```

Comment:
`e2e_node` tests always run on the same host which runs the patched kubelet, this is by design and I don't think it will ever change. So I'd just read the related files in `/sys` saving `exec.Command`s

I think the only distinction that matters is:
- do we need the host/system view? IOW, what the test expects? if so we can just access the host data as I wrote above
- do we need the container view? IOW, what the container actually got? If so the safest and better way is to exec into the container and report which resource it got.

It seems to me we are in the first case now

Reply:
You're right. I was influenced too much by how we use `lscpu` to detect NUMA nodes, cores per socket etc. than I didn't think of just using `os.ReadFile` 😄 will change it.